File: tests/bench/bench.js
function ehPrimo(n) {
    if (n < 2) return false;
    if (n === 2) return true;
    if (n % 2 === 0) return false;
    
    const limite = Math.floor(Math.sqrt(n));
    for (let i = 3; i <= limite; i += 2) {
        if (n % i === 0) return false;
    }
    return true;
}

function contarPrimos(limite) {
    let contador = 0;
    for (let numero = 2; numero <= limite; numero++) {
        if (ehPrimo(numero)) contador++;
    }
    return contador;
}

function main() {
    const limite = 1000000;
    
    // Warm-up para V8
    contarPrimos(1000);
    
    const resultado = contarPrimos(limite);
    console.log(`RESULT:${resultado}`);
}

main();

File: tests/bench/bench.py
import math

def eh_primo(n):
    if n < 2: return False
    if n == 2: return True
    if n % 2 == 0: return False
    
    # Otimização: usar int ao invés de float
    limite = int(math.sqrt(n)) + 1
    for i in range(3, limite, 2):
        if n % i == 0: return False
    return True

def contar_primos(limite):
    contador = 0
    for numero in range(2, limite + 1):
        if eh_primo(numero):
            contador += 1
    return contador

def main():
    limite = 1000000
    
    # Warm-up para JIT
    contar_primos(1000)
    
    resultado = contar_primos(limite)
    print(f"RESULT:{resultado}")

if __name__ == "__main__":
    main()
